src/vs/workbench/api/common/extHostStatusBar.ts

```diff
@@ -334,10 +355,22 @@ export class ExtHostStatusBar implements ExtHostStatusBarShape {
 		}
 	}
 
+	async $provideTooltip(entryId: string, cancellation: vscode.CancellationToken): Promise<string | IMarkdownString | undefined> {
+		const entry = this._entries.get(entryId);
+		if (!entry) {
+			return undefined;
+		}
+
+		return MarkdownString.fromStrict(await entry.tooltip2?.(cancellation));
```

Comment:
I would await the `entry.tooltip2` call and then check again for cancellation and then return `undefined` if cancelled.

Reply:
Fixed.